ci: release process

jacderida opened the pull request:
The release process now uses the `release-plz` tool.

The version bumping workflow uses a small script that runs `release-plz update` and parses its output to determine which crates had version bumps. A commit is then generated based on that. Unlike `smart-release`, `release-plz update` doesn't create a commit or tags. So now the version bumping is only producing the commit and pushing that to `main`.

The release workflow is then kicked off, which builds the binaries, then uses the `release-plz release` command to publish crates that were bumped using `update`. The same command also performs Github Releases for the binary crates and creates tags for every crate that was published.

Finally, the binary assets are attached to the releases and uploaded to S3.

## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 07 Jun 23 18:44 UTC
This pull request introduces various changes, including additions and updates to GitHub Actions workflows and release processes. It adds new environment variables, updates 'on' triggers, and modifies the release-plz.toml file. Additionally, it introduces a new `Justfile` with tasks for building and releasing artifacts across different platforms.
<!-- reviewpad:summarize:end -->

oetyng commented on src/routing/core/messaging.rs:
```diff
@@ -250,7 +253,11 @@ impl Core {
         self.send_message_for_dst_accumulation(
             src_prefix.name(),
-            DstLocation::DirectAndUnrouted(*self.section.chain().last_key()),
+            // DstLocation::Section(*self.section.chain().last_key()),
```
```suggestion
```

oetyng replied:
First I thought these things were some commented code remnants, but are they supposed to be docs?

oetyng replied:
```suggestion
```

oetyng commented on src/routing/routing_api/tests/mod.rs:
```diff
@@ -350,9 +364,14 @@ async fn aggregate_proposals() -> Result<()> {
     let wire_msg = WireMsg::single_src(
         &nodes[THRESHOLD],
-        DstLocation::DirectAndUnrouted(*section.chain().last_key()),
+        // DstLocation::Section(*section.chain().last_key()),
```
```suggestion
```

oetyng replied:
```suggestion
```

oetyng commented on src/routing/routing_api/tests/mod.rs:
```diff
@@ -999,7 +1018,11 @@ async fn handle_bounced_untrusted_message() -> Result<()> {
     let bounced_wire_msg = WireMsg::single_src(
         &other_node,
-        DstLocation::DirectAndUnrouted(pk1),
+        // DstLocation::Node(pk1),
```
```suggestion
```

oetyng replied:
```suggestion
```

oetyng commented on src/routing/routing_api/tests/mod.rs:
```diff
@@ -1106,7 +1129,11 @@ async fn handle_sync() -> Result<()> {
     let wire_msg = WireMsg::single_src(
         &old_node,
-        DstLocation::DirectAndUnrouted(pk1),
+        // DstLocation::Node(pk1),
```
```suggestion
```

jacderida committed:
ci: release process

The release process now uses the `release-plz` tool.

The version bumping workflow uses a small script that runs `release-plz update` and parses its
output to determine which crates had version bumps. A commit is then generated based on that. Unlike
`smart-release`, `release-plz update` doesn't create a commit or tags. So now the version bumping is
only producing the commit and pushing that to `main`.

The release workflow is then kicked off, which builds the binaries, then uses the `release-plz
release` command to publish crates that were bumped using `update`. The same command also performs
Github Releases for the binary crates and creates tags for every crate that was published.

Finally, the binary assets are attached to the releases and uploaded to S3.
